protos/flyteidl/datacatalog/datacatalog.proto

```diff
@@ -93,6 +95,26 @@ message ListDatasetsResponse {
     string next_token = 2;
 }
 
+message AcquireLeaseRequest {
```

Comment:
So datacatalog and lease semantics look weird. I think we should alter the existing API to provide optimistic locking - reservations.

The API is invoked from flytepropeller here - https://github.com/flyteorg/flytepropeller/blob/master/pkg/controller/nodes/task/catalog/datacatalog/datacatalog.go#L115

How about we create a new GetArtifact to be
GetOrReserveArtifact ?